### PR TITLE
Impart Security's WAF and API security plugin

### PR DESCRIPTION
**Pull request by marc-harrison**
### Description

Adds Impart Security's plugin to Kong's hub.

### CODE REVIEW

File: app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Kong Plugin Impart
+publisher: Impart Security
+
+categories:
+  - security
```
[the remark this answers is not in the document]
Comment (marc-harrison) [reply app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml:5]: Just want to double check.  The guide says use -
```
# Uncomment "community" if you are submitting a 3rd party plugin that isn't a technical partner. 
# No other categories apply if "community" is active.
# Uncomment "third-party-partner" if you are a Kong technical partner, 
# and choose a secondary category from the list.
```
Should I only use `third-party-partner` or `community`? I'm going to assume `community` and also set:
```
techpartner: false
```

Let me know if it should be something else.  Thanks!

[the remark this answers is not in the document]
Comment (DaniellaFreese) [reply app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml]: right, with the new changes, `community` is the label to use.

File: app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Kong Plugin Impart
```
Comment (DaniellaFreese) [inline app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml:1]: Suggestion: change the name to `Impart Security` 

I would recommend removing "kong plugin" from the name. This is the field that will be used to create the title on the card.

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+3. Restart {{site.base_gateway}}:
+
+    ```sh
+    kong restart
+    ```
+
+## Using the plugin
+
+First of all, make sure you have installed an Impart Inspector. Steps to do that can be found on the Impart console [Integrations](https://console.impartsecurity.net/orgs/_/integrations?q=kong) page.
```
Comment (DaniellaFreese) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:55]: Suggestion: this plugin requires having installed an impart inspector

Comment (DaniellaFreese) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:55]: Suggestion: this plugin requires the Impart Inspector to be installed. Navigate to the Impart console for step-by-step instructions (url).

File: app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml (hunk 2 above)
Comment (marc-harrison) [reply app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml]: Thanks!

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md (hunk 3 above)
Comment (marc-harrison) [reply app/_hub/impart-security/kong-plugin-impart/overview/_index.md]: Thanks!  Updated.

**Comment by lena-larionova** [comment]
Preview link works now: https://deploy-preview-7672--kongdocs.netlify.app/hub/impart-security/kong-plugin-impart/

File: app/assets/images/icons/hub/impart_kong-plugin-impart.png
Comment (lena-larionova) [inline app/assets/images/icons/hub/impart_kong-plugin-impart.png:1]: Can you please change the filename to `impart-security_kong-plugin-impart.png`? It has to match the exact name of the folder to get picked up by the generator.

File: app/_hub/impart-security/kong-plugin-impart/_changelog.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+## Changelog
+
+**kong-plugin-impart 0.2.3**
+
+* Official release of kong-plugin-impart.
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/_changelog.md:5]:
```suggestion
* Official release of `kong-plugin-impart`. 
```

File: app/_hub/impart-security/kong-plugin-impart/_changelog.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+## Changelog
+
+**kong-plugin-impart 0.2.3**
+
+* Official release of kong-plugin-impart. 
+This plugin is compatible with {{site.base_gateway}} 3.4.x, 3.5.x, 3.6.x and 3.7.x at release time.
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/_changelog.md:6]:
```suggestion
This plugin is compatible with {{site.base_gateway}} 3.4.x, 3.5.x, 3.6.x, and 3.7.x at release time.
```

File: app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+name: Impart Security
+publisher: Impart Security
+
+categories:
+  - community
+
+type: plugin
+
+desc: Integrate Impart Security's WAF and API security protection platform with Kong.
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/_metadata/_index.yml:9]:
```suggestion
desc: Integrate Impart Security's WAF and API security protection platform with Kong
```

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
+---
+nav_title: Overview
+title: Overview
+---
+
+Impart's API Protection and WAF platform delivers comprehensive protection for APIs, microservices, and serverless applications in cloud-native environments.
+
+Use the Impart plugin to
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:8]:
```suggestion
Use the Impart plugin to:
```

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+* Discover and catalog your API and web application Attack Surface
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:9]:
```suggestion
* Discover and catalog your API and web application Attack Surface.
```

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+* Protect your APIs and web applications from injection, enumeration, automated threats, and other attacks.
+* Find and help you fix your API and web application vulnerabilities and misconfigurations with built in API testing
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:11]:
```suggestion
* Find and help you fix your API and web application vulnerabilities and misconfigurations with built in API testing.
```

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+     -F "config.inspector_rpc_addr=<inspector_rpc_addr>"
+```
+
+### {{site.konnect_product_name}}
+
+- Depending on where you want to enable Impart, select `Plugins`
+- Click on `+ New Plugin`
+- On `Custom Plugins`, select `Kong Plugin Impart`
+- Fill the `Inspector Rpc Addr` field
+- Hit `Save`
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/overview/_index.md:73]:
```suggestion
If you are a {{site.konnect_short_name}} administrator, install the Impart plugin as a [custom plugin](/konnect/gateway-manager/plugins/add-custom-plugin/).

If the plugin has already been installed by an administrator, you can enable it through {{site.konnect_short_name}}:
1. Depending on where you want to enable Impart, select **Plugins**.
2. Click on **+ New Plugin**
3. On **Custom Plugins**, select **Kong Plugin Impart**.
4. Fill in the **Inspector Rpc Addr** field.
5. Click **Save**.
```
Depending on the user's role, they'll have a different relationship with this plugin. It won't be available in Konnect by default.

**Review by lena-larionova — CHANGES_REQUESTED** [review]
Just have some minor feedback around formatting + need the icon filename to change.

**Comment by lena-larionova** [comment]
Please ignore the failing smoke test, it doesn't work on forks

File: app/assets/images/icons/hub/impart_kong-plugin-impart.png
Comment (marc-harrison) [reply app/assets/images/icons/hub/impart_kong-plugin-impart.png:1]: Updated.  Thanks for the feedback/suggestions!

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md (hunk 10 above)
Comment (marc-harrison) [reply app/_hub/impart-security/kong-plugin-impart/overview/_index.md]: The linter doesn't like `Rpc` or `Addr`.  Previously, I bypassed it by using \`Inspector Rpc Addr\`.  Did you have a preference on how to handle the vale spelling failure?  Add Rpc/Addr to the dictionary? Switch it back to `Inspector Rpc Addr`?  I'lll change it to **\`Inspector Rpc Addr\`** for the time being.  Reopen if you'd prefer something else.  Thanks!

**Comment by marc-harrison** [comment]
@lena-larionova I think this should be ready.  The Broken Links seems to be something on netlify's needing to be redeployed.  Let me know if there's anything else.  Thanks!

File: app/_hub/impart-security/kong-plugin-impart/overview/_index.md (hunk 10 above)
Comment (lena-larionova) [reply app/_hub/impart-security/kong-plugin-impart/overview/_index.md]: Adding it to the dictionary is the way to go, yep.

**Comment by lena-larionova** [comment]
Looks good to me! 
Holding off on publishing until we get the green light from our end.

**Review by lena-larionova — APPROVED** [review]
@marc-harrison We've received the go-ahead to publish this. 
Do you want to add 3.8 compatibility as well?

**Comment by marc-harrison** [comment]
@lena-larionova added.  Thanks!

File: app/_hub/impart-security/kong-plugin-impart/_changelog.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
+## Changelog
+
+**kong-plugin-impart 0.2.3**
+
+* Official release of `kong-plugin-impart`. 
+This plugin is compatible with {{site.base_gateway}} 3.4.x, 3.5.x, 3.6.x, and 3.7.x at release time.
```
Comment (lena-larionova) [inline app/_hub/impart-security/kong-plugin-impart/_changelog.md:6]:
```suggestion
This plugin is compatible with {{site.base_gateway}} 3.4.x, 3.5.x, 3.6.x, 3.7.x, and 3.8.x at release time.
```

**Pull request merged**